tests/common/include/tests/common/test_plugin.inl

```diff
@@ -0,0 +1,293 @@
+/** Detray library, part of the ACTS project (R&D line)
+ * 
+ * (c) 2020 CERN for the benefit of the ACTS project
+ * 
+ * Mozilla Public License Version 2.0
+ */
+
+#include "common/types.hpp"
+
+#include <cmath>
+#include <climits>
+
+#include <gtest/gtest.h>
+
+/// @note __plugin has to be defined with a preprocessor command
+using namespace algebra;
+
+// Two-dimensional definitions
+using point2cart = __plugin::cartesian2::point2;
```

Comment:
I think you branched off a bit too early, in a newer version of `detray` I have removed the strongly typed attempt that makes a difference between `cartesian::point2` and `polar::point2`.

I moved all `point3, vector3, point2, vector2` out into the `__plugin` namespace, this makes the code way simpler as you don't have to deduce the type from the local representation nor the `transform_type`.

I also removed the transform templating quite a bit.

Reply:
Ah, damn. I will have to rework parts of the detray integration then.